tasks/password-generator/src/main.rs

```diff
@@ -0,0 +1,172 @@
+            break;
+        }
+    }
+    // you convert the vector of characters into a string
+    // using the turbofish syntax
+    base_password.iter().collect::<String>()
+}
+// validator function for our first argument
+// picked 15 as the length must be greater than
+// our randomly generated range above (10)
+fn is_valid_length(value: String) -> Result<(), String> {
+    let length = value.parse::<u32>();
+    // if we cannot parse the string into an i32
+    if length.is_err() {
+        Err(String::from("Please provide valid password"))
+    } else if length.unwrap() < 15 {
+        Err(String::from("Mininum password length is 15"))
+    } else {
+        Ok(())
+    }
+}
+fn main() {
+    // create our new CLI
+    // clap provides powerful defaults so we don't have to
+    // write all the logic here
+    // For example, clap takes care of the help and version flags by default
+    let matches = App::new("password-generator")
+        .version("0.1")
+        .about("generate a password according to the rosetta code rules: http://rosettacode.org/wiki/Password_generator")
+        // configure our first required argument
+        .arg(Arg::with_name("LENGTH")
+             .help("password length")
+             // make it SECURE by default
+             // https://cheatsheetseries.owasp.org/cheatsheets/Password_Storage_Cheat_Sheet.html
+             .default_value("160")
+             .required(true)
+             .validator(is_valid_length)
+             .index(1)
+             .takes_value(true)
+            )
+        // configure our second required argument
+        .arg(Arg::with_name("COUNT")
+             .help("how many passwords to generate")
+             .default_value("1")
+             .required(true)
+             .index(2)
+             .takes_value(true)
+            )
+        .get_matches();
+
+    // It's safe to call unwrap because the value with either be what the user provides
+    // at runtime or our defaults configured above
+    let parsed_count: i32 = matches.value_of("COUNT").unwrap().parse().unwrap();
+    let parsed_length: usize = matches.value_of("LENGTH").unwrap().parse().unwrap();
+    // we don't need the index since we'll print to standard out
+    // in rust, underscores indicate unused variables
+    for _ in 0..parsed_count {
+        println!("{}", generate_password(parsed_length));
+    }
+}
+
+// declare a module for testing
+// keep it within this file for convenience
+#[cfg(test)]
+mod tests {
+    use super::generate_password;
+
+    // test our core password logic according to the rules described in the wiki
+    #[test]
+    fn generate_password_customizes_length() {
+        let a_password = generate_password(50);
+        assert_eq!(a_password.len(), 50);
+    }
+    #[test]
+    fn generate_password_has_numerals() {
+        // TODO how can I pass in char::is_ascii_digit instead of defining a closure?
+        assert!(generate_password(50).chars().any(|c| c.is_ascii_digit()));
```

Comment:
I had tried ` assert!(generate_password(50).chars().any(char::.is_ascii_digit));` as in an [exercism task](https://exercism.io/tracks/rust/exercises/bob/solutions/de25567cbcda4fb298f3d4b2c398f2fb) but got compiler errors:
```
error[E0631]: type mismatch in function arguments
   --> tasks/password-generator/src/main.rs:114:47
    |
114 |         assert!(generate_password(50).chars().any(char::is_ascii_digit));
    |                                               ^^^
    |                                               |
    |                                               expected signature of `fn(char) -> _`
    |                                               found signature of `for<'r> fn(&'r char) -> _`
```

Reply:
You could use `copied()` on the iterator but I think it's fine as is.